BaseListener/UseCase/DirectDebitProcessUseCase.cs

```diff
@@ -0,0 +1,38 @@
+using BaseListener.Gateway.Interfaces;
+using System.Threading.Tasks;
+using BaseListener.UseCase.Interfaces;
+using Amazon.Lambda.APIGatewayEvents;
+using System.Text.Json;
+using BaseListener.Boundary.Response;
+using BaseListener.Helpers.GeneralModels;
+using BaseListener.Factories;
+using System.Net;
+
+namespace BaseListener.UseCase
+{
+    public class DirectDebitProcessUseCase : IDirectDebitProcessUseCase
+    {
+        private readonly IHttpApiGateway _gateway;
+
+        public DirectDebitProcessUseCase(IHttpApiGateway gateway)
+        {
+            _gateway = gateway;
+        }
+
+        public async Task<APIGatewayProxyResponse> ProcessExecuteAsync(APIGatewayProxyRequest apiGatewayProxyRequest)
+        {
+            APIGatewayProxyResponse response = await _gateway.GetAsync(apiGatewayProxyRequest).ConfigureAwait(false);
+
+            if (response.StatusCode == (int) HttpStatusCode.OK)
+            {
+                PaginatedResponse<TransactionResponse> model = JsonSerializer.Deserialize<PaginatedResponse<TransactionResponse>>(response.Body);
```

Comment:
Given the expected data type is already specified as a generic method type parameter, the line 28 could probably be simplified to `var model = .... `

Reply:
This comments is made abosolete by the one bellow